feg/gateway/services/n7_n40_proxy/n7/model_conversion.go

```diff
@@ -0,0 +1,567 @@
+		accessType = sbi.AccessTypeNON3GPPACCESS
+	default:
+		return nil
+	}
+	return &accessType
+}
+
+func getSbiTimeZone(tzProto *protos.Timezone) *sbi.TimeZone {
+	if tzProto == nil {
+		return nil
+	}
+	offsetMin := tzProto.GetOffsetMinutes()
+	isNegative := false
+	if offsetMin < 0 {
+		isNegative = true
+		offsetMin = -offsetMin
+	}
+	tz := fmt.Sprintf("%02d:%02d", offsetMin/60, offsetMin%60)
+	if isNegative {
+		tz = "-" + tz
+	} else {
+		tz = "+" + tz
+	}
+	tzStr := sbi.TimeZone(tz)
+	return &tzStr
+}
+
+func getSbiPduSesionType(pduSessionType protos.PduSessionType) sbi.PduSessionType {
+	switch pduSessionType {
+	case protos.PduSessionType_IPV4:
+		return sbi.PduSessionTypeIPV4
+	case protos.PduSessionType_IPV4IPV6:
+		return sbi.PduSessionTypeIPV4V6
+	case protos.PduSessionType_IPV6:
+		return sbi.PduSessionTypeIPV6
+	case protos.PduSessionType_UNSTRUCTURED:
+		return sbi.PduSessionTypeUNSTRUCTURED
+	default:
+		return sbi.PduSessionTypeIPV4
+	}
+}
+
+func getSbiBool(val *bool) bool {
+	if val == nil {
+		return false
+	}
+	return *val
+}
+
+func getSbiUinteger(val *sbi.Uinteger) uint32 {
+	if val == nil {
+		return 0
+	}
+	return uint32(*val)
+}
+
+func GetPccRules(
+	policyDecision *n7_sbi.SmPolicyDecision,
+) ([]*protos.StaticRuleInstall, []*protos.DynamicRuleInstall, []string) {
+	var (
+		staticRules   []*protos.StaticRuleInstall
+		dynamicRules  []*protos.DynamicRuleInstall
+		rulesToRemove []string
+	)
+
+	if policyDecision.PccRules == nil {
+		return staticRules, dynamicRules, rulesToRemove
+	}
+
+	for ruleId, pccRule := range policyDecision.PccRules.AdditionalProperties {
+		if pccRule.PccRuleId == "" {
+			// This rule to be removed
+			rulesToRemove = append(rulesToRemove, ruleId)
+			continue
+		}
+		activationTime, deactivationTime := GetActivateDeactivationTime(policyDecision, &pccRule)
+		policyRule := GetPolicyRuleProto(policyDecision, &pccRule)
+		if policyRule == nil {
+			// This is a static rule. Static rule only has pccRuleId and RefCondData
+			staticRule := &protos.StaticRuleInstall{
+				RuleId:           ruleId,
+				ActivationTime:   activationTime,
+				DeactivationTime: deactivationTime,
+			}
+			staticRules = append(staticRules, staticRule)
+			continue
+		}
+
+		dynamicRule := &protos.DynamicRuleInstall{
+			PolicyRule:       policyRule,
+			ActivationTime:   activationTime,
+			DeactivationTime: deactivationTime,
+		}
+		dynamicRules = append(dynamicRules, dynamicRule)
+	}
+
+	return staticRules, dynamicRules, rulesToRemove
+}
+
+func GetActivateDeactivationTime(
+	policyDecision *n7_sbi.SmPolicyDecision,
+	pccRule *n7_sbi.PccRule,
+) (activationTime, deactivationTime *timestamp.Timestamp) {
+	if pccRule.RefCondData != nil && *pccRule.RefCondData != "" {
+		condData, found := policyDecision.Conds.Get(*pccRule.RefCondData)
+		if found {
+			activationTime = ConvertToProtoTimeStamp(condData.ActivationTime)
+			deactivationTime = ConvertToProtoTimeStamp(condData.DeactivationTime)
+		}
+	}
+	return
+}
```

Comment:
If possible, and it makes sense, try to have as less indentation levels as possible. It makes the code easier le read.

See how in this code you don't have to jump back and forth to understand, It is lineal

```
func GetActivateDeactivationTime(policyDecision *n7_sbi.SmPolicyDecision, pccRule *n7_sbi.PccRule,
) (activationTime, deactivationTime *timestamp.Timestamp) {
	if pccRule.RefCondData != nil && *pccRule.RefCondData == "" {
		return
	}
	condData, found := policyDecision.Conds.Get(*pccRule.RefCondData)
	if found == false {
		return
	} 
	activationTime = ConvertToProtoTimeStamp(condData.ActivationTime)
	deactivationTime = ConvertToProtoTimeStamp(condData.DeactivationTime)
}
```

Reply:
Done